src/practice.js

```diff
@@ -34,7 +34,23 @@ export default class Practice extends React.Component {
     this.toggleAnswer = this.toggleAnswer.bind(this)
     this.previousCard = this.previousCard.bind(this)
     this.nextCard = this.nextCard.bind(this)
+    this.handleDifficulty = this.handleDifficulty.bind(this)
   }
+  handleDifficulty(event) {
+    const { flashcards, saveFlashcardDifficulty } = this.props
+    const { currentCardIndex } = this.state
+    if (event.target.id === 'easy') {
+      flashcards[currentCardIndex].handleDifficulty = 'easy'
```

Comment:
It's not a good idea to mutate state directly. You'll need to make a copy of the array and the object you're updating.